pytest_sphinx.py

```diff
@@ -71,6 +71,7 @@ def _is_doctest(config, path, parent):
 
 _DIRECTIVE_RE = re.compile(
     r"""
+    (?P<myst_directive>`{3}{eval-rst}\n?)?
```

Comment:
Therefore, this line, doesn't currently add change anything.

If you want that `_DIRECTIVE_RE` finds  (for markdown content) a match for the following input 

```md
```{{eval-rst}}
.. docstest::

```

but not for 

```md
```{{xxxxx}}
.. doctest::

```
we have to extend the code that calls  `_DIRECTIVE_RE.match`

Reply:
I've been having some issues when testing this locally in my project, so this may be why.

I will look into this.

(note to self, the above message refers to [how the regex checks every line](https://github.com/thisch/pytest-sphinx/pull/31/files#r921581003)

Reply:
Could you take a stab at getting it to work? (using the existing parser)

I am going try a fork of this branch that uses the myst-parser/docutils/etc API. 

<details>
<summary>Here's what you can do:</summary>

I added a test to `tests/test_sphinx_doctest.py::TestDirectives::test_doctest`. It automatically skips if myst-parser is not found

`pip install myst-parser` or `pip install -e .[myst-parser]` (bash) or `pip install -e ".[myst-parser]"` (zsh)

Then `py.test`

**Re-invoke on file rerun:**

`pip install pytest-watcher`

_With this test_

`py.test tests/test_sphinx_doctest.py::TestDirectives::test_doctest ; ptw . tests/test_sphinx_doctest.py::TestDirectives::test_doctest`

_Project-wide_

`py.test -x -s -v; ptw . -x -s -v`

</details>

Reply:
@thisch 

Could you take a stab at getting it to work? (using the existing parser)

I am going try a fork of this branch that uses the myst-parser/docutils/etc API. 

<details>
<summary>Here's what you can do:</summary>

I added a test to `tests/test_sphinx_doctest.py::TestDirectives::test_doctest`. It automatically skips if myst-parser is not found

`pip install myst-parser` or `pip install -e .[myst-parser]` (bash) or `pip install -e ".[myst-parser]"` (zsh)

Then `py.test`

**Re-invoke on file rerun:**

`pip install pytest-watcher`

_With this test_

`py.test tests/test_sphinx_doctest.py::TestDirectives::test_doctest ; ptw . tests/test_sphinx_doctest.py::TestDirectives::test_doctest`

_Project-wide_

`py.test -x -s -v; ptw . -x -s -v`

</details>